contrib/runners/go_runner/go_runner.py

```diff
@@ -0,0 +1,380 @@
+        except Exception:
+
+            error_msg = "Error checking that Go is installed. Please ensure a valid Go " \
+                        "installation exists on the system and is present in the PATH"
+
+            LOG.error(error_msg)
+            raise Exception(error_msg)
+
+        #TODO(mierdin): Add directory to GOPATH seamlessly 
+        #export GOPATH=$GOPATH:$(pwd)/actions
+
+        #You will also need to add the resulting bin/ directory to path, or refer to it directly
+
+        LOG.debug('Checking entry_point')
+        if not self.entry_point:
+            # /opt/stackstorm/packs/influxdb/actions/go_example.go
+            error_msg = 'Action "%s" is missing entry_point attribute' % (self.action.name)
+            LOG.error(error_msg)
+            raise Exception(error_msg)
+        LOG.debug('entry_point is %s' % self.entry_point)
+
+        # User has structured project in "go gettable" format. Respect this.
+        #
+        # TODO(mierdin): This isn't the most attractive way of detecting this choice. However,
+        # because of the necessary precautions in st2/st2common/st2common/content/utils.py,
+        # function "get_pack_file_abs_path", an exact file must be provided to the entry_point
+        # field in action metadata.
+        #
+        # This means, if we have a "go get" style structure in place, the path we must use (if
+        # we are to know which binary this action intends to use) must look like this:
+        #
+        # src/vendor_example/cmd/print_cyan/main.go
+        #
+        # It might be worth revisiting this - as this is obviously not ideal UX, but I can deal
+        # with it for the sake of getting a PoC out there for this runner.
+        if self.entry_point[-8:] == "/main.go":
+            entry_parent = os.path.abspath(os.path.dirname(self.entry_point) + "/..")
+
+            # LOG.debug("entry_parent - %s" % entry_parent)
+
+            env = os.environ.copy()
+            LOG.debug("GOPATH - %s" % env['GOPATH'])
+
+            if entry_parent.split("/")[-1] != "cmd":
+                error_msg = "Go project not structured correctly - cannot compile"
+                LOG.error(error_msg)
+                raise Exception(error_msg)
+
+            install_param = entry_parent + "/..."
+            self.binary_name = os.path.dirname(self.entry_point).split("/")[-1]
+
+        # Treat the entry point as a random script - compile this file directly
+        else:
+            install_param = self.entry_point
+            self.binary_name = os.path.splitext(basename(self.entry_point))[0]
+
+        LOG.debug("Compiling program")
+        try:
+            args = [
+                "go",
+                "install",
+                install_param
+            ]
+            LOG.debug("Running command '%s'" % " ".join(args))
+            exit_code, stdout, stderr, timed_out = run_command(cmd=args, stdout=subprocess.PIPE,
```

Comment:
Not sure how I feel about us being responsible for the compilation step. At least not for every run. Needs more discussion.